lib/ansiblelint/rules/PackageHasRetryRule.py

```diff
@@ -67,6 +67,16 @@ class PackageHasRetryRule(AnsibleLintRule):
         "zypper",
     ]
 
+    _module_ignore_states = [
+        "absent",
+    ]
+
     def matchtask(self, file, task):
-        return (task['action']['__ansible_module__'] in self.package_modules
-                and 'until' not in task)
+        module = task["action"]["__ansible_module__"]
+        if module in self._package_modules:
+            if 'until' not in task:
+                # Exclude tasks where state is `absent`.
+                if task['action'].get('state') not in self._module_ignore_states:
+                    message = '{0} {1}'
+                    return message.format(self.shortdesc, module)
```

Comment:
Oh, the return value should be bool. It's strongly discouraged to mix different return types in a function.

Reply:
For better or worse, currently ansible-lint rules use mixed return types, returning True ansible-lint cli outputs the `shortdesc` field, returning text ansible-lint cli outputs the text. In this case True might be descriptive enough for this rule.